src/main/java/com/boomi/flow/services/boomi/mdh/match/MatchResponseMapper.java

```diff
@@ -4,96 +4,93 @@
 
-        var propertiesMatched = new Property(FuzzyMatchDetialsConstants.MATCH, new ArrayList<>());
-        var propertiesDuplicated = new Property(FuzzyMatchDetialsConstants.DUPLICATE, new ArrayList<>());
-        var propertiesAlreadyLinked = new Property(FuzzyMatchDetialsConstants.ALREADY_LINKED, new ArrayList<>());
-        var object = new MObject(universeId + " match", externalId, Arrays.asList(propertiesMatched, propertiesDuplicated, propertiesAlreadyLinked));
-        object.setTypeElementBindingDeveloperName(object.getDeveloperName());
+        var propertiesMatched = new Property(FuzzyMatchDetailsConstants.MATCH, new ArrayList<>());
+        var propertiesDuplicated = new Property(FuzzyMatchDetailsConstants.DUPLICATE, new ArrayList<>());
+        var propertiesAlreadyLinked = new Property(FuzzyMatchDetailsConstants.ALREADY_LINKED, new ArrayList<>());
 
-        var properties =  new ArrayList<Property>();
-        result.getEntity().get(universe.getName())
-                .forEach((key, value) -> properties.add(new Property(key, (String) value)));
+        if ("SUCCESS".equals(result.getStatus())) {
+            var matches = result.getMatch().stream()
+                    .map(match -> createMatchesToProperty(universe, match, universe.getIdField(), true))
+                    .collect(Collectors.toList());
 
-        properties.add(new Property(GoldenRecordConstants.SOURCE_ID_FIELD, result.getIdResource()));
-        properties.add(new Property(FuzzyMatchDetialsConstants.FUZZY_MATCH_DETAILS, (MObject) null));
-        properties.addAll(object.getProperties());
+            propertiesMatched = new Property(FuzzyMatchDetailsConstants.MATCH, matches);
 
-        object.setProperties(properties);
+            var duplicates = result.getDuplicate().stream()
+                    .map(duplicate -> createMatchesToProperty(universe, duplicate, universe.getIdField(), true))
+                    .collect(Collectors.toList());
+
+            propertiesDuplicated = new Property(FuzzyMatchDetailsConstants.DUPLICATE, duplicates);
 
-        if ("SUCCESS".equals(result.getStatus())) {
-            result.getMatch().forEach(match -> addMatchesToProperty(propertiesMatched, universe, match, universe.getIdField(), true));
-            result.getDuplicate().forEach(match -> addMatchesToProperty(propertiesDuplicated, universe, match, universe.getIdField(), true));
         } else if ("ALREADY_LINKED".equals(result.getStatus())) {
-            var entityH = result.getEntity().get(universe.getName());
-            addAlreadyLinked(propertiesAlreadyLinked, universe, entityH, universe.getIdField());
+            var entityLinked = result.getEntity().get(universe.getName());
+            propertiesAlreadyLinked = new Property(FuzzyMatchDetailsConstants.ALREADY_LINKED,
+                    Arrays.asList(createAlreadyLinked(universe, entityLinked, universe.getIdField())));
```

Comment:
I don't think you need to use `Arrays.asList` here as `Property` should have a constructor that takes a single `MObject`